### PR TITLE
fix: Handle BLIP-2 model output format

### PR DESCRIPTION
**Pull request by thisisiron**
# What does this PR do?

I have modified the code to return logits in addition to loss.

Instead of:

```
return {"loss": loss}
```

The code now returns:
```
return {
    "loss": loss,
    "logits": logits
}
```

Output:
```
Done!
First values of original logits: tensor([[ 1.9322,  1.9379,  7.4007],
        [-1.4741, -1.1189,  8.6593],
        [-1.4213, -1.2489,  6.1976]])
First values of HF logits: tensor([[ 1.9322,  1.9379,  7.4007],
        [-1.4741, -1.1189,  8.6593],
        [-1.4213, -1.2489,  6.1976]])
Looks ok!
Generating a caption...
Original generation: ['merlion']
HF generation: ['Question: what object is in this image? Answer: merlion']
```

Fixes # ([issue](https://github.com/huggingface/transformers/issues/34704#issue-2652956392))

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@amyeroberts, @qubvel

### CODE REVIEW

**Comment by LysandreJik** [comment]
In case you have bandwidth to review the conversion script fix @NielsRogge

**Comment by Rocketknight1** [comment]
gentle ping @NielsRogge
